### PR TITLE
Connect group interactions and poll creation to api

### LINKED ISSUE #30: Implement interactions: vote/prepare

### LINKED ISSUE #110: Create poll page is missing name field

### PR DESCRIPTION
**Pull request by g-wizzy**
Features a massive refactor on the groups page.

Close #110 close #30

#82 and #82 could be closed by this, but the API needs to be tested.

### CODE REVIEW

File: resources/js/components/pages/create-poll.vue [hunk 1]
```diff
@@ -40,6 +60,127 @@ export default {
...
   },
-  middleware: "auth"
+  middleware: "auth",
+  data() {
+    return {
+      groups: [],
+      selectedGroup: {},
+      poll: {
+        name: "",
+        users: [{ name: localStorage.getItem("username") + " (You)" }],
```
Comment (g-wizzy) [inline resources/js/components/pages/create-poll.vue]: Forgot to remove this default value. It is overridden on page load anyway. I'll remove it along with other corrections if there are any.

File: app/Http/Resources/GroupResource.php [hunk 2]
```diff
@@ -18,6 +18,7 @@ public function toArray($request)
             'id' => $this->id,
             'name' => $this->name,
             'users' => UserResource::collection($this->whenLoaded('users')),
+            'userCount' => $this->users()->count(),
```
Comment (Blatoy) [inline app/Http/Resources/GroupResource.php]: Please use snake case

File: resources/js/components/groups/add-group-modal.vue [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      group: placeholderGroup
+    };
+  },
+  methods: {
+    open: function() {
+      this.group = placeholderGroup;
+
+      // Set the user as unremovable entry in list
+      this.$refs.userList.unremovableItems.push({
+        id: localStorage.getItem("id"),
```
Comment (Blatoy) [inline resources/js/components/groups/add-group-modal.vue:58]: Do we have id in localstorage?

File: resources/js/components/groups/add-group-modal.vue [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+          this.$parent.groups.push(this.group);
+        })
+        .catch(error => {
+          alert("ERROR: Could not save group");
+        });
+    },
+    close: function() {},
+    addUser: function(email) {
+      this.axios
+        .get("/users/by-mail/" + email)
```
Comment (Blatoy) [inline resources/js/components/groups/add-group-modal.vue]: Missing `encodedURIComponent` for the email

File: resources/js/components/groups/edit-group-modal.vue [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+          this.$refs.modal.active = true;
+        })
+        .catch(error => {
+          alert("ERROR: Could not load group");
+        });
+    },
+    save: function() {
+      this.axios.patch("/groups/" + this.group.id, {
+        name: this.group.name,
+        users_to_add: this.usersToAdd.map(user => user.id),
```
Comment (Blatoy) [inline resources/js/components/groups/edit-group-modal.vue]: users_to_add should receive
`[{"id": <id>, "admin": <bool>}]`
users_to_remove and items_to_remove should receive
`[{"id": <id>}]`

File: resources/js/components/pages/create-poll.vue [hunk 6]
```diff
@@ -40,6 +60,127 @@ export default {
...
+      this.poll.users = [this.user];
+      this.selectedGroup.users.forEach(user => {
+        if (user.id !== localStorage.getItem("id")) {
+          this.poll.users.push(user);
+        }
+      });
+    },
+    addUser: function(email) {
+      this.axios
+        .get("/users/by-mail/" + email)
```
Comment (Blatoy) [inline resources/js/components/pages/create-poll.vue]: Missing `encodeURIComponent`

**Review by Blatoy — CHANGES_REQUESTED** [review]

File: resources/js/components/groups/add-group-modal.vue (hunk 3 above)
Comment (g-wizzy) [reply resources/js/components/groups/add-group-modal.vue:58]: Yes we do

**Review by Blatoy — APPROVED** [review]

**Pull request merged**